### PR TITLE
Show bucket region during submit confirmation [BW-449]

### PR DESCRIPTION
**Pull request by cjllanwarne**
Tested running locally pointed at the `alpha` environment.

Example submit confirmation dialog:

~removed~ (see updates in comment below)

NB: styling suggestions welcome!

### CODE REVIEW

File: src/pages/workspaces/workspace/workflows/LaunchAnalysisModal.js
Comment (aednichols) [inline src/pages/workspaces/workspace/workflows/LaunchAnalysisModal.js]: Users can specify custom output locations, right?

The following is more concise and possibly more accurate by not over-promising:
```suggestion
        'Workspace bucket:',
```

Comment (cjllanwarne) [reply src/pages/workspaces/workspace/workflows/LaunchAnalysisModal.js]: I didn't think that they could? At least not in Terra?

Comment (cjllanwarne) [reply src/pages/workspaces/workspace/workflows/LaunchAnalysisModal.js]: (at least, at least not via this submit dialog?)

Comment (aednichols) [reply src/pages/workspaces/workspace/workflows/LaunchAnalysisModal.js]: I could write a task that uploads a localized file to an arbitrary bucket, I'm pretty sure I've heard of people doing this or similar somewhere

It's not likely to happen in Terra, OTOH making statements about where data always goes is a commitment

File: src/components/RegionFlaggedContent.js
Comment (aednichols) [inline src/components/RegionFlaggedContent.js]:
```suggestion
    (location === 'EUROPE-NORTH1') ? { flag: '🇫🇮', aka: 'Finland' } :
```

**Comment by cjllanwarne** [comment]
From conversation with Joy:

* Show the region name, not the bucket name on this page.
* No need for the copy-to-clipboard link.
* [Optional] Show the region flag and bucket name on the "Outputs" tab instead

File: src/components/RegionFlaggedContent.js
Comment (evanbernstein) [inline src/components/RegionFlaggedContent.js]: TOL. I worry about the tech debt of this block of code. The flags are nice and cute, but also now maintenance that we are creating for ourselves, that we have to remember to fix everytime we add a new region (which isn't in/near this code to remind us). I'd like a cleaner "failure" state that at least displays the text of the region. It might not be pretty, but would still display something to the user rather than nothing.

Comment (aednichols) [reply src/components/RegionFlaggedContent.js]: +1

IMO we should display only the region, with `monospaced font styling`

File: src/pages/workspaces/workspace/workflows/LaunchAnalysisModal.js
Comment (aednichols) [inline src/pages/workspaces/workspace/workflows/LaunchAnalysisModal.js]: I still believe we should reword this to make the promise less specific and drop the disclaimer.

```suggestion
        'Data written to the workspace bucket resides in region `region-name`'
```

File: src/pages/workspaces/workspace/workflows/LaunchAnalysisModal.js
Comment (aednichols) [inline src/pages/workspaces/workspace/workflows/LaunchAnalysisModal.js]: Based on my experience working with our end-users, people will fall into three categories:

80% people who don't know they can do this
15% people who are aware this is a possibility but never do it
5% people who actually write data somewhere custom

For the 80% who don't know what this means, it kind of sounds like Terra is failing to protect their data from mysterious "external actions" that could move files somewhere else at any time ("not guaranteed to remain only in this bucket").

```suggestion
```

**Review by aednichols — CHANGES_REQUESTED** [review]
Sorry to re-gull (did I just coin that?) but I think we have to be very careful with language that's going to be read (and attempted to be understood) by the entire cross-section of the workflows userbase

**Comment by cjllanwarne** [comment]
Updates following feedback (inc. from @aednichols, @evanbernstein and Joy):

#### Launch modal:

* We now list the region rather than the bucket, as per the original ticket
* Wording has been updated to be closer in wording (and spirit) to the contents of the "Outputs" tab

image


* Hovering over the help tooltip icon yields:

image

* Unrecognized regions will now still show up (eg if I "unrecognize" the US multi-region:

image

#### Outputs tab

* Added new content in the form of the bucket region, and made it collapsed by default to avoid taking up too much screen space. 
* Made `Files` into a link back to the Data pages (since I didn't realize initially that this was what it was referencing)

image

File: src/pages/workspaces/workspace/workflows/LaunchAnalysisModal.js
Comment (cjllanwarne) [reply src/pages/workspaces/workspace/workflows/LaunchAnalysisModal.js]: I'm reluctant to switch to this wording.

1. I think 'Data written to the workspace bucket resides in region X' ignores the most important part of the story w.r.t. submitting a workflow - namely that this workflow will do some writing of data to that bucket.
1. I think "your outputs will be saved to X region" is true, even if it's not the full story.
1. I prefer optimizing the message for the common case and providing disclaimers for power users rather than making the message less useful in order to cover edges cases which the 80% will never have to worry about.

I'll also make the weak-sauce "resort to prior art" argument that the "Outputs" tab already has a form of words which is essentially identical to the statement being made here.

File: src/pages/workspaces/workspace/workflows/LaunchAnalysisModal.js
Comment (cjllanwarne) [reply src/pages/workspaces/workspace/workflows/LaunchAnalysisModal.js]: Have re-worded. Let me know if the new form rings truer.

File: src/pages/workspaces/workspace/Data.js
Comment (cjllanwarne) [inline src/pages/workspaces/workspace/Data.js]: Doing this felt hacky. @panentheos / @zarsky-broad any ideas on ways to do it better?

Comment (zarsky-broad) [reply src/pages/workspaces/workspace/Data.js]: We have an example somewhere of loading making a url including a query with something specific to load first and then removing that from the shown url once we do, I just can't remember where.

**Review by aednichols — COMMENTED** [review]
Latest iteration looks great to me, will hold off with a full review until Kyle's requested changes land

File: src/pages/workspaces/workspace/Data.js
Comment (cjllanwarne) [reply src/pages/workspaces/workspace/Data.js]: or is it possible to pass props to the constructor directly when you call a reference link?

File: src/components/RegionFlaggedContent.js
Comment (panentheos) [inline src/components/RegionFlaggedContent.js]: I'm not sure about the value of this abstraction as-is. Even if we do end up with another caller in the other PR, they're different enough that they would probably be cleaner as separate code paths. We can always extract out smaller shared bits, as we discover them.

File: src/components/RegionFlaggedContent.js
Comment (panentheos) [inline src/components/RegionFlaggedContent.js]: Since this copy button only applies to the other caller, I think we should keep it inline there and not try to extract it.

File: src/pages/workspaces/workspace/Data.js
Comment (panentheos) [reply src/pages/workspaces/workspace/Data.js]: Instead of using a path param, we could add an (optional) query param that the page consumes, to initialize the selected data type. That way we could remove it after the page loads so it doesn't become part of the permanent URL scheme.

File: src/pages/workspaces/workspace/workflows/LaunchAnalysisModal.js
Comment (panentheos) [inline src/pages/workspaces/workspace/workflows/LaunchAnalysisModal.js]: Why do we need the fallback `|| {}` here?

File: src/pages/workspaces/workspace/workflows/LaunchAnalysisModal.js
Comment (panentheos) [inline src/pages/workspaces/workspace/workflows/LaunchAnalysisModal.js]: We should add some linebreaks and indentation here to make this more readable.

File: src/pages/workspaces/workspace/workflows/LaunchAnalysisModal.js
Comment (panentheos) [inline src/pages/workspaces/workspace/workflows/LaunchAnalysisModal.js]: This isn't valid, if we want flexbox it should be `display: 'flex'`

File: src/pages/workspaces/workspace/workflows/LaunchAnalysisModal.js
Comment (panentheos) [inline src/pages/workspaces/workspace/workflows/LaunchAnalysisModal.js]: Right now this isn't accessible to keyboard or screen reader users. As a general rule, use the `Clickable` component or one of its derivatives to get the proper interactivity. In this case, there's an `InfoBox` component that might be a better fit here.

File: src/pages/workspaces/workspace/workflows/WorkflowView.js
Comment (panentheos) [inline src/pages/workspaces/workspace/workflows/WorkflowView.js]: nit: This `Fragment` and the one below are unnecessary, we could just include the content directly.

File: src/pages/workspaces/workspace/workflows/WorkflowView.js
Comment (panentheos) [inline src/pages/workspaces/workspace/workflows/WorkflowView.js]: The horizontal margin is pushing this out of alignment with the rest of the content, we should probably just have vertical margins.

File: src/pages/workspaces/workspace/workflows/WorkflowView.js
Comment (panentheos) [inline src/pages/workspaces/workspace/workflows/WorkflowView.js]: If this tooltip contains important information, then we should include some accessible text so screen readers can see it as well.

File: src/pages/workspaces/workspace/workflows/WorkflowView.js
Comment (panentheos) [inline src/pages/workspaces/workspace/workflows/WorkflowView.js]: This flex wrapper isn't doing much as-is. If the intention was to preserve the alignment between the icon and the 'Files' text, then these styles should go on the `Link` instead, and this wrapper would be unnecessary.

File: src/pages/workspaces/workspace/workflows/WorkflowView.js
Comment (panentheos) [inline src/pages/workspaces/workspace/workflows/WorkflowView.js]: nit: We should probably keep this margin consistent with the 'Files' one above. Also, this should be a `flex` so the icon and the text are aligned.

File: src/pages/workspaces/workspace/workflows/WorkflowView.js
Comment (panentheos) [inline src/pages/workspaces/workspace/workflows/WorkflowView.js]: TOL: This seems like a nice change overall, but given the complexity with regard to the Data page, perhaps this could have been split out as a separate PR.

File: src/components/RegionFlaggedContent.js
Comment (cjllanwarne) [reply src/components/RegionFlaggedContent.js]: I think it possibly made more sense before I extracted out the region functions (because there was one case where I couldn't re-use this component anyway). Separate code paths seems logical to me.

File: src/pages/workspaces/workspace/workflows/LaunchAnalysisModal.js
Comment (panentheos) [inline src/pages/workspaces/workspace/workflows/LaunchAnalysisModal.js]: nit: Is 'delocalize' a commonly understood word, or could we say something more straightforward like 'write'?

**Review by panentheos — APPROVED** [review]
Looks good, thanks for splitting this out. Tested locally.

File: src/pages/workspaces/workspace/workflows/LaunchAnalysisModal.js
Comment (salonishah11) [inline src/pages/workspaces/workspace/workflows/LaunchAnalysisModal.js:126]: What is the purpose of `Loading...` here? Is it like if the region hasn't been instantiated it displays `Loading` till then?

Comment (salonishah11) [reply src/pages/workspaces/workspace/workflows/LaunchAnalysisModal.js:126]: Never mind. Please ignore the question. I tested it locally and found out the purpose of `Loading`. It is exactly what I said ^

**Review by salonishah11 — APPROVED** [review]
LGTM 👍  I have tested it locally and it works as expected

File: src/pages/workspaces/workspace/workflows/LaunchAnalysisModal.js
Comment (cjllanwarne) [reply src/pages/workspaces/workspace/workflows/LaunchAnalysisModal.js:126]: It's so that if the API call to google hasn't replied yet we can still display the rest of the content around it and leave a placeholder for the region information.

**Comment by cjllanwarne** [comment]
Ignoring flaky integration test failure from known issues.

**Pull request merged**
